cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/client/FleetShardObservabilityClient.java

```diff
@@ -0,0 +1,216 @@
+package org.bf2.cos.fleetshard.sync.client;
+
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.bf2.cos.fleetshard.sync.FleetShardSyncConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.observability.v1.Observability;
+import com.redhat.observability.v1.ObservabilitySpec;
+import com.redhat.observability.v1.observabilityspec.ConfigurationSelector;
+import com.redhat.observability.v1.observabilityspec.SelfContained;
+import com.redhat.observability.v1.observabilityspec.Storage;
+import com.redhat.observability.v1.observabilityspec.selfcontained.*;
+import com.redhat.observability.v1.observabilityspec.storage.Prometheus;
+import com.redhat.observability.v1.observabilityspec.storage.prometheus.VolumeClaimTemplate;
+import com.redhat.observability.v1.observabilityspec.storage.prometheus.volumeclaimtemplate.Spec;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.Subscription;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.SubscriptionBuilder;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.SubscriptionSpecBuilder;
+
+@ApplicationScoped
+public class FleetShardObservabilityClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FleetShardObservabilityClient.class);
+
+    @Inject
+    KubernetesClient kubernetesClient;
+    @Inject
+    FleetShardSyncConfig config;
+
+    private String observabilityNamespace = "";
+
+    @PostConstruct
+    public void init() {
+        observabilityNamespace = config.observability().namespace();
+    }
+
+    public void setupObservability() {
+        if (!config.observability().enabled()) {
+            LOGGER.warn("Observability is not enabled.");
+            return;
+        }
+
+        var observabilityCRD = kubernetesClient.resources(CustomResourceDefinition.class)
+            .withName("observabilities.observability.redhat.com")
+            .get();
+        if (observabilityCRD == null) {
+            LOGGER.error("Observability CustomResourceDefinition is not present. Observability will not be enabled.");
+            return;
+        }
+
+        if (observabilityNamespace.isBlank()) {
+            LOGGER.error("Observability namespace is not set!");
+            return;
+        }
+
+        if (config.observability().subscription().enabled()) {
```

Comment:
What's the use case where a user sets `enabled` to `true` and `subscription.enabled` to false?

Reply:
If/when we are able to tell the observability-operator to install the prometheus operator in a different namespace. Then we can (not mandatory) just disable the Subscription and install as a dependency.